In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
!pip install bolides
from bolides import BolideDataFrame;

C:\Users\under\anaconda3\Lib\site-packages\lightkurve\prf\__init__.py:7: UserWarning: Warning: the tpfmodel submodule is not available without oktopus installed, which requires a current version of autograd. See #1452 for details.
  warnings.warn(


In [3]:
bdf = BolideDataFrame(source='glm')
bdf

downloading: 22.7MiB [00:00, 31.4MiB/s]


,datetime,longitude,latitude,source,detectedBy,confidenceRating,lightcurveStructure,peak_energy_g16,peak_energy_g17,peak_energy_g18,peak_energy_cat_g16,peak_energy_cat_g17,peak_energy_cat_g18,_id,status,name,description,latitudeDelta,longitudeDelta,lastModifiedBy,lastModifiedDate,createdBy,createdDate,submittedBy,submittedDate,publishedBy,publishedDate,howFound,groundTrack,nearbyLightningActivity,otherDetectingSources,otherInformation,humanClassification,reason,rejectedBy,rejectedDate,duration,beginningAltitude,attachments,isManuallyGenerated,isPipelineIssue,files,location,checkout_time,assessmentScore,images,csv,platform,pdfFilenames,geometry,phase,moon_fullness,solarhour,sun_alt_obs,sun_alt_app,date_retrieved
0,2025-06-20 20:18:58.875000+00:00,-112.922256,-8.272419,glm,GLM-18,medium,good,NaN,NaN,0.0,,,Faint,6856de0e3493a5c6a21a7b8a,PUBLISHED,,,0.000000,0.040627,jdotson,2025-06-27T22:11:00.070Z,auto,2025-06-21T09:30:06.374Z,rlongenb,2025-06-25T21:09:50.032Z,jdotson,2025-06-27T22:11:00.070Z,{'GLM-18': ['forest']},"{'GLM-18': {'category': 'Multi-Pixel', 'value'...",isolated,,,Bolide,,,None,0.048065,89.43417,[{'location': {'coordinates': [-112.9628829956...,False,False,[G18_2025-06-20_20h18m58.875_-8.272_-112.961_1...,"{'coordinates': [-112.92225646972656, -8.27241...",-2,"{'score': 0.894256055355072, 'assessment': 'ca...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G18_s20251712018400_e2025...,GLM-18,[G18_2025-06-20_20h18m58.875_-8.272_-112.961_1...,POINT (-112.92226 -8.27242),0.843339,0.313323,12.759668,56.398641,56.388109,2025-07-07 22:29:55.326444
1,2025-06-20 17:33:06.317000+00:00,-168.461510,-37.292299,glm,GLM-18,low,good,NaN,NaN,0.0,,,Faint,6856de0e3493a5c6a21a7b87,PUBLISHED,,,0.039610,0.058907,jdotson,2025-06-27T22:11:14.727Z,auto,2025-06-21T09:30:06.368Z,rlongenb,2025-06-25T21:08:44.285Z,jdotson,2025-06-27T22:11:14.727Z,{'GLM-18': ['forest']},"{'GLM-18': {'category': 'Multi-Pixel', 'value'...",isolated,,,Bolide,,,None,0.274281,,[{'location': {'coordinates': [-168.5204162597...,False,False,[G18_2025-06-20_17h33m06.323_-37.325_-168.473_...,"{'coordinates': [-168.46150970458984, -37.2922...",-2,"{'score': 0.948855459690094, 'assessment': 'ca...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G18_s20251711733000_e2025...,GLM-18,[G18_2025-06-20_17h33m06.323_-37.325_-168.473_...,POINT (-168.46151 -37.2923),0.839409,0.321182,6.292664,-10.668599,-10.668599,2025-07-07 22:29:55.326444
2,2025-06-20 13:48:15.181000+00:00,-160.275452,20.025345,glm,GLM-18,low,minimal,NaN,NaN,0.0,,,Faint,6856de0e3493a5c6a21a7b89,PUBLISHED,,,0.000000,0.000000,jdotson,2025-06-27T22:11:29.353Z,auto,2025-06-21T09:30:06.372Z,rlongenb,2025-06-25T21:08:01.631Z,jdotson,2025-06-27T22:11:29.353Z,{'GLM-18': ['forest']},"{'GLM-18': {'category': 'Single-Pixel', 'value...",isolated,,,Bolide,,,None,0.040055,,[{'location': {'coordinates': [-160.2754516601...,False,False,[G18_2025-06-20_13h48m15.181_20.025_-160.275_1...,"{'coordinates': [-160.27545166015625, 20.02534...",-2,"{'score': 0.9447590112686157, 'assessment': 'c...","[{'name': 'Map', 'url': '/img/OR_GLM-L2-LCFA_G...",[/csv/OR_GLM-L2-LCFA_G18_s20251711348000_e2025...,GLM-18,[G18_2025-06-20_13h48m15.181_20.025_-160.275_1...,POINT (-160.27545 20.02534),0.834082,0.331836,3.091482,-27.295634,-27.295634,2025-07-07 22:29:55.326444
3,2025-06-20 12:34:32.739000+00:00,-128.397560,33.464287,glm,GLM-18,medium,good,NaN,NaN,0.0,,,Faint,6856de0e3493a5c6a21a7b86,PUBLISHED,,,0.000000,0.001015,jdotson,2025-06-27T22:12:27.437Z,auto,2025-06-21T09:30:06.366Z,rlongenb,2025-06-25T21:07:10.990Z,jdotson,2025-06-27T22:12:27.437Z,{'GLM-18': ['forest']},"{'GLM-18': {'category': 'Single-Pixel', 'value...",isolated,,,Bolide,,,None,0.061418,86.30142,[{'location': {'coordinates': [-128.3985748291...,False,False,[G18_2025-06-20_12h34m32.742_33.464_-128.399_1...,"{'coordinates': [-128.3975601196289, 33.464286...",-2,"{'score': 0.92606121301651, 'assessment': 'can...","[{'name': 'Map', 'ur

In [4]:
highbdf= bdf[bdf.confidenceRating == "high"]

In [5]:
latdelta=highbdf.latitudeDelta #isolate latitude and longitude change 
longdelta=highbdf.longitudeDelta

In [6]:
print(np.max(latdelta), np.max(longdelta))

15.681482315063477 18.090576171875


In [7]:
delta= np.sqrt(latdelta**2+longdelta**2) #use the pythagorean theorem to compute the actual track length of the bolides

In [8]:
np.max(delta)

23.941132676380626

In [9]:
bolidelist=[] #list of the longest tracked bolides 
latdeltalist=[] #the corresponding change in latitude for the longest tracked bolides
longdeltalist=[] #corresponding change in longitude, this will be used later to find the bolide locations in highbdf
indexlist=[] #tracks the placement of the bolides selected in delta

for i in range(len(delta)):   
    if delta.iloc[i]>3: #arbitrary number that shows the bolides with the greatest delta
        bolidelist.append(delta.iloc[i]) #append all values to their corresponding lists 
        latdeltalist.append(latdelta.iloc[i])
        longdeltalist.append(longdelta.iloc[i])
        indexlist.append(i)

bolidelist

[23.941132676380626,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.605551275463989,
 3.605551275463989,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.605551275463989,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 3.1622776601683795,
 4.47213595499958,
 3.1622776601683795]

In [10]:
len(latdeltalist)

24

In [11]:
bolidenumbers=[] #will hold the locations in highbdf of the bolides we want
w=0 
for i in range(len(highbdf)):
    for e in range (len(latdeltalist)): #search latdeltalist and longdeltalist for the corresponding values...
        if latdelta.iloc[i] == latdeltalist[e] and longdelta.iloc[i] == longdeltalist[e] and i not in bolidenumbers: #... and compare those values to make sure that specific bolide is the one we want
             bolidenumbers.append(i) #save the index 

We need to remove duplicates so we have 15 bolides. I'm going to take out 9 who have the value 3.16..., the indexes of which are 1, 2, 6, 8, 9, 10, 11, 12, 14

In [12]:
bolidenumbers #pre removal- len=24

[77,
 218,
 402,
 431,
 440,
 469,
 479,
 480,
 484,
 488,
 507,
 509,
 510,
 529,
 568,
 581,
 593,
 596,
 624,
 629,
 631,
 639,
 659,
 668]

In [13]:
bolidenumbers.remove(bolidenumbers[1]) #remove some shorter tracked bolides so our sample size is 15
bolidenumbers.remove(bolidenumbers[2]) #this removal is ONLY for the sample size, it's possible the 3D trajectories for these are quite interesting
bolidenumbers.remove(bolidenumbers[6]) #but for now they didn't make the cut
bolidenumbers.remove(bolidenumbers[8]) 
bolidenumbers.remove(bolidenumbers[9])
bolidenumbers.remove(bolidenumbers[10])
bolidenumbers.remove(bolidenumbers[11])
bolidenumbers.remove(bolidenumbers[12])
bolidenumbers.remove(bolidenumbers[14])

In [14]:
totalbolidelist=[]

for i in range (len(bolidenumbers)):
    totalbolidelist.append(highbdf.iloc[bolidenumbers[i]]) #make a list of the information regarding the bolides we want

In [15]:
bolidenumbers #final location of the 15 selected bolides within highbdf

[77, 402, 440, 469, 479, 480, 488, 507, 510, 568, 593, 624, 631, 639, 668]

In [16]:
totalbolidelist #all the information regarding every bolide selected

[datetime                                    2024-09-19 09:52:50.049000+00:00
 longitude                                                         -95.642914
 latitude                                                          -11.445276
 source                                                                   glm
 detectedBy                                                            GLM-16
 confidenceRating                                                        high
 lightcurveStructure                                                very good
 peak_energy_g16                                                          0.0
 peak_energy_g17                                                          NaN
 peak_energy_g18                                                          NaN
 peak_energy_cat_g16                                                   Bright
 peak_energy_cat_g17                                                         
 peak_energy_cat_g18                                            